controller/src/main/java/io/pravega/controller/server/bucket/ZooKeeperBucketManager.java

```diff
@@ -73,25 +88,37 @@ public void startBucketOwnershipListener() {
                     break;
                 case CHILD_REMOVED:
                     int bucketId = Integer.parseInt(ZKPaths.getNodeFromPath(event.getData().getPath()));
-                    RetryHelper.withIndefiniteRetriesAsync(() -> tryTakeOwnership(bucketId),
-                            e -> log.warn("{}: exception while attempting to take ownership for bucket {}: {}", getServiceType(),
-                                    bucketId, e.getMessage()), getExecutor());
+                    bucketStore.getBucketsForController(processId, getServiceType())
```

Comment:
Do we still need this method? As per my understanding this method is a listener which is used to update the znode with bucket services. This was useful in the previous approach. Since you already are updating the map and have added a listener to that map, this looks redundant to me. You should remove this method and put `stop` and `start` services on any instance of controller in indefinite retryable block.

Reply:
that way we will be able to get rid of redundant listener calls and associated processes since we do not need this action with the new design anymore

Reply:
Yes, its making listener call redundant. Removed the previous listener in the current approach.